src/Sources/Code/Module.cpp

```diff
@@ -234,71 +276,125 @@ void WriteSimVar(SimVar& simVar, Client* client) {
 		sizeof(simVar.Value),
 		&simVar.Value
 	);
-	if (hr != S_OK) {
-		fprintf(stderr, "MobiFlight[%s]: Error on Setting Client Data. %u, SimVar: %s (ID: %u)", client->Name.c_str(), hr, simVar.Name.c_str(), simVar.ID);
-	}
+
+		if (hr != S_OK) {
+			fprintf(stderr, "MobiFlight[%s]: Error on Setting Client Data. %lu, SimVar: %s (ID: %u)", client->Name.c_str(), hr, simVar.Name.c_str(), simVar.ID);
+		}
 #if _DEBUG
-	std::cout << "MobiFlight[" << client->Name.c_str() << "]: SimVar " << simVar.Name.c_str();
+	std::cout << "MobiFlight[" << client->Name.c_str() << "]: Written SimVar " << simVar.Name.c_str();
 	std::cout << " with ID " << simVar.ID << " has value " << simVar.Value << std::endl;
 #endif
 }
 
-// Register a single SimVar and send the current value to SimConnect Clients 
-void RegisterSimVar(const std::string code, Client* client) {
+// Register a single SimVar and send the current value to SimConnect Clients
+void RegisterSimVar(const std::string code, Client* client, const bool isString) {
```

Comment:
I am thinking that it would be nice to either refactor the method so that it calls internal RegisterFloatSimVar and RegisterStringSimVar

or, that the code parts with if(isString) {... string specific code} else {... float specific parts} are extracted into their own respective methods. Ideally, we could even put that into another class file.

That would improve code readiability.

Reply:
Wilco :)